pym/bob/generators/VisualStudioCode.py

```diff
@@ -0,0 +1,175 @@
+
+# global template for the whole workspace file
+JSON_WORKSPACE_TEMPLATE = {
+    "folders": [],
+    "launch": {
+        "configurations": [],
+        "compunds": []
+    },
+    "tasks": {
+        "version": "2.0.0",
+        "options": {
+            "shell" : {
+                "executable": "",
+                "args": []
+            },
+            "cwd": ""
+        },
+        "tasks": [],
+        "inputs": [ {
+            "id": "buildArguments",
+            "type": "promptString",
+            "description": "Please give 'bob dev' arguments (see 'bob dev --help' for more information)",
+            "default": "-v"
+        } 
+        ]
+    },
+    "settings": {
+        "C_Cpp.default.includePath" : [],
+        "C_Cpp.default.defines": []
+    }
+}
+
+JSON_TASK_TEMPLATE = {
+    "label": "",
+    "type" : "shell",
+    "command": "",
+    "args": [""],
+    "problemMatcher": [
+        "$msCompile",
+        "$gcc"
+    ],
+    "group": {
+        "kind": "build",
+        "isDefault": True
+    }
+}
+
+TASK_LIST = [
+    {"label" :"Bob dev", "args": ["-v"] },
+    {"label" :"Bob dev (force)", "args": ["-v", "-f"] },
+    {"label" :"Bob dev (no checkout)", "args": ["-v", "-b"] },
+    {"label" :"Bob dev (no deps)", "args": ["-v", "-n"] },
+    {"label" :"Bob dev (no checkout, no deps)", "args": ["-v", "-nb"] },
+    {"label" :"Bob dev (no checkout, clean)", "args": ["-v", "-b", "--clean"] },
+    {"label" :"Bob dev (ask for arguments)", "args": ["${input:buildArguments}"] },
+]
+
+def getCorrectPathList(recipesRoot, pathlist):
+    if len(pathlist) == 0: return []
+    if isMsys():
+        values = os.popen('cygpath -w {}'.format(" ".join([quoteBash(i) for i in pathlist]))).read().strip().split("\n")
+        return [str(recipesRoot.joinpath(PureWindowsPath(i))) for i in values]
+    else:
+        return [str(recipesRoot.joinpath(i)) for i in pathlist]
+
+def getCorrectPath(recipesRoot, path):
+    if isMsys() and len(path) > 0:
+        return  recipesRoot.joinpath(PureWindowsPath(os.popen('cygpath -w {}'.format(quoteBash(path))).read().strip()))
+    else:
+        return recipesRoot.joinpath(path)
+
+class Project:
+    def __init__(self, recipesRoot, scan, includeList):
+        self.isRoot = scan.isRoot
+        self.packagePath = scan.stack
+        self.workspacePath =  getCorrectPath(recipesRoot, scan.workspacePath)
+        self.headers = getCorrectPathList(recipesRoot, scan.headers)
+        self.sources = getCorrectPathList(recipesRoot, scan.sources)
+        self.resources = getCorrectPathList(recipesRoot, scan.resources)
+        self.incPaths =  getCorrectPathList(recipesRoot, scan.incPaths)
+        self.dependencies = scan.dependencies
+        self.runTargets = getCorrectPathList(recipesRoot, scan.runTargets)
+        includeList += self.incPaths
+
+class VsCodeGenerator(CommonIDEGenerator):
+    def __init__(self):
+        super().__init__("vsCode", "Generate Visual Studio Code workspace")
```

Comment:
```suggestion
        super().__init__("vscode", "Generate Visual Studio Code workspace")
```

Otherwise the command line help is wrong.